packages/cdk/resources/CognitoFunctions.ts

```diff
@@ -0,0 +1,237 @@
+
+    // define some variables that we need if we are doing mock authorization
+    let mockTokenLambda: LambdaFunction
+
+    // set up things for mock authorization
+    if (props.useMockOidc) {
+      if (props.mockOidcjwksEndpoint === undefined ||
+        props.mockOidcUserInfoEndpoint === undefined ||
+        props.mockOidcTokenEndpoint === undefined ||
+        props.mockOidcClientId === undefined ||
+        props.mockOidcIssuer === undefined
+      ) {
+        throw new Error("Attempt to use mock oidc but variables are not defined")
+      }
+
+      // secret used by mock token lambda that holds the JWT private key
+      const mockJwtPrivateKey = new Secret(this, "MockJwtPrivateKey", {
+        secretName: `${props.stackName}-mockJwtPrivateKey`,
+        secretStringValue: SecretValue.unsafePlainText("ChangeMe"),
+        encryptionKey: jwtKmsKey
+      })
+      const getMockJWTPrivateKeySecret = new ManagedPolicy(this, "getMockJWTPrivateKeySecret", {
+        statements: [
+          new PolicyStatement({
+            actions: [
+              "secretsmanager:GetSecretValue"
+            ],
+            resources: [
+              mockJwtPrivateKey.secretArn
+            ]
+          })
+        ]
+      })
+
+      // lambda for mock token endpoint
+      mockTokenLambda = new LambdaFunction(this, "MockTokenResources", {
+        serviceName: props.serviceName,
+        stackName: props.stackName,
+        lambdaName: `${props.stackName}-mockToken`,
+        additionalPolicies: [
+          props.tokenMappingTableWritePolicy,
+          props.tokenMappingTableReadPolicy,
+          props.useTokensMappingKmsKeyPolicy,
+          useJwtKmsKeyPolicy,
+          getMockJWTPrivateKeySecret
+        ],
+        logRetentionInDays: props.logRetentionInDays,
+        packageBasePath: "packages/cognito",
+        entryPoint: "src/token.ts",
+        lambdaEnvironmentVariables: {
+          idpTokenPath: props.mockOidcTokenEndpoint,
+          TokenMappingTableName: props.tokenMappingTable.tableName,
+          UserPoolIdentityProvider: props.mockPoolIdentityProviderName,
+          oidcjwksEndpoint: props.mockOidcjwksEndpoint,
+          jwtPrivateKeyArn: mockJwtPrivateKey.secretArn,
+          userInfoEndpoint: props.mockOidcUserInfoEndpoint,
+          useSignedJWT: "false",
+          oidcClientId: props.mockOidcClientId,
+          oidcIssuer: props.mockOidcIssuer
+        }
+      })
+    }
+
+    // secret used by token lambda that holds the JWT private key
+    const primaryJwtPrivateKey = new Secret(this, "PrimaryJwtPrivateKey", {
+      secretName: `${props.stackName}-primaryJwtPrivateKey`,
+      secretStringValue: SecretValue.unsafePlainText("ChangeMe"),
+      encryptionKey: jwtKmsKey
+    })
+    primaryJwtPrivateKey.addToResourcePolicy( new PolicyStatement({
+      effect: Effect.ALLOW,
+      principals: [props.deploymentRole],
+      actions: [
+        "secretsmanager:PutSecretValue"
+      ],
+      resources:["*"]
```

Comment:
as above

Reply:
it is not meaningless - it is required. See resources section of https://docs.aws.amazon.com/kms/latest/developerguide/key-policy-overview.html#key-policy-elements